src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryAdder.java

```diff
@@ -202,7 +202,7 @@ static List<URL> retrieve(@NonNull LibraryRecord record, @NonNull LibraryRetriev
         FilePath libDir = new FilePath(execution.getOwner().getRootDir()).child("libs/" + record.getDirectoryName());
         Boolean shouldCache = cachingConfiguration != null;
         final FilePath versionCacheDir = new FilePath(LibraryCachingConfiguration.getGlobalLibrariesCacheDir(), record.getDirectoryName());
-        ReentrantReadWriteLock retrieveLock = getReadWriteLockFor(record.getDirectoryName());
+        ReentrantReadWriteLock retrieveLock = getReadWriteLockFor(record.getName());
```

Comment:
This is to make it consistent across accesses because record.getName() is no longer equal to record.getDirectoryName().